packages/language-picker/src/language-picker.tsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import React, { useState } from 'react';
+import { useI18n } from '@automattic/react-i18n';
+
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type LanguageGroup = {
+	id: string;
+	name: () => string;
+	subTerritories?: string[];
+	countries?: string[];
+	default?: boolean;
+};
+
+type LanguageSlug = string;
+type WPLocale = string;
+
+type BaseLanguage = {
+	langSlug: LanguageSlug;
+	name: string;
+	popular?: number;
+	rtl?: boolean;
+	territories: string[];
+	value: number;
+	wpLocale: WPLocale | '';
+};
+
+type SubLanguage = BaseLanguage & { parentLangSlug: string };
+
+export type Language = BaseLanguage | SubLanguage;
+
+type Props = {
+	onSelectLanguage: ( language: Language ) => void;
+	languages: Language[];
+	languageGroups: LanguageGroup[];
+};
+
+const LanguagePicker = ( { onSelectLanguage, languages, languageGroups }: Props ) => {
```

Comment:
Optional: you can fix the eslint warning about defining module boundaries by using the `React.FC` type.

```suggestion
const LanguagePicker: React.FC< Props > = ( { onSelectLanguage, languages, languageGroups } ) => {
```